lib/services/workflow-api-service.js

```diff
@@ -143,11 +143,13 @@ function workflowApiServiceFactory(
             return taskGraphProtocol.cancelTaskGraph(instanceId);
         })
         .then(function(cancelledGraphId) {
-            if (cancelledGraphId !== instanceId) {
-                // Cancelled graph ID doesn't match the requested ID.
-                throw new Errors.TaskCancellationError('Failed to cancel ' + graphId);
-            }
-            return waterline.graphobjects.needByIdentifier(graphId);
+            return waterline.graphobjects.needByIdentifier(graphId)
+            .then (function(workflow) {
+                if (workflow.active()) {
+                    throw new Errors.TaskCancellationError('Failed to cancel ' + graphId);
+                }
+                return waterline.graphobjects.needByIdentifier(graphId);
```

Comment:
I think similar to the random 1.1 functional test errors,  for some reason the underlining cancelledGraphId doesn't always get set by cancelTaskGraph, it ends up undefined.